### PR TITLE
Refactor rule config

### PR DESCRIPTION
**Pull request by HerringtonDarkholme**
## Release Notes

* **Refactor**
  * Updated rewriter configuration to separate matching rules from fix instructions.
  * Revised outline rule schema so shared rewriters are now defined at the outline common level.

* **Bug Fixes / Validation**
  * Improved validation around rewriter fix definitions, including clearer errors for missing/undefined fix components and related hint handling.

* **Tests**
  * Expanded coverage for rewriter fix parsing and validation scenarios, including additional ignored cases for edge-condition errors.

### CODE REVIEW

File: crates/config/src/transform/rewrite.rs [hunk 1]
```diff
@@ -149,12 +155,10 @@ fn make_edit<D: Doc>(
 #[cfg(test)]
 mod test {
   use super::*;
-  use crate::check_var::CheckHint;
-  use crate::from_str;
   use crate::rule::DeserializeEnv;
   use crate::rule::referent_rule::RuleRegistration;
-  use crate::rule_core::SerializableRuleCore;
   use crate::test::TypeScript;
+  use crate::{SerializableRewriter, from_str};
```
[the remark this answers is not in the document]
Comment (HerringtonDarkholme) [reply crates/config/src/transform/rewrite.rs:161]: fixed

File: crates/config/src/check_var.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 use crate::RuleCore;
 use crate::fixer::{Fixer, FixerError};
+use crate::rewriter::Rewriter;
```
[the remark this answers is not in the document]
Comment (HerringtonDarkholme) [reply crates/config/src/check_var.rs:3]: wil fix later
